src/java/eu/rssw/rcode/ClassCompilationUnit.java

```diff
@@ -0,0 +1,217 @@
+public class ClassCompilationUnit {
+    @XmlAttribute
+    public String packageName, className, inherits;
+    @XmlElement
+    public List<String> interfaces = new ArrayList<String>();
+    @XmlAttribute
+    public boolean isAbstract, isFinal, isInterface, isEnum;
+
+    @XmlElement(name = "classComment")
+    public List<String> classComment = new ArrayList<String>();
+    @XmlElement(name = "constructor")
+    public List<Constructor> constructors = new ArrayList<Constructor>();
+    @XmlElement(name = "method")
+    public List<Method> methods = new ArrayList<Method>();
+    @XmlElement(name = "property")
+    public List<Property> properties = new ArrayList<Property>();
+    @XmlElement(name = "event")
+    public List<Event> events = new ArrayList<Event>();
+    @XmlElement(name = "using")
+    public List<Using> usings = new ArrayList<Using>();
+    @XmlElement(name = "member")
+    public List<EnumMember> enumMembers = new ArrayList<>();
+
+    public void classToXML(File out) throws JAXBException, IOException {
+        FileOutputStream fos = new FileOutputStream(out);
+        JAXBContext context = JAXBContext.newInstance(this.getClass().getPackage().getName());
+        Marshaller marshaller = context.createMarshaller();
+        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+        marshaller.marshal(this, fos);
+        fos.close();
+    }
+
+    public void merge(ClassCompilationUnit unit) {
+        classComment.addAll(unit.classComment);
+
+        // Fusion des commentaires pour les propriétés
+        for (Property prop : properties) {
+            for (Property prop2 : unit.properties) {
+                if (prop.name.equalsIgnoreCase(prop2.name))
+                    prop.propertyComment = prop2.propertyComment;
+            }
+        }
+        // Fusion des commentaires pour les événements
+        for (Event evt : events) {
+            for (Event evt2 : unit.events) {
+                if (evt.eventName.equalsIgnoreCase(evt2.eventName))
+                    evt.eventComment = evt2.eventComment;
+            }
+        }
+        // Fusion des commentaires pour les constructeurs
+        for (Constructor c : constructors) {
+            for (Constructor c2 : unit.constructors) {
+                if (hasSameSignature(c.parameters, c2.parameters)) {
+                    c.constrComment = c2.constrComment;
+                }
+            }
+        }
+        // Fusion des commentaires pour les méthodes
+        for (Method m2 : unit.methods) {
+            boolean found = false;
+            for (Method m : methods) {
+                if (m.methodName.equalsIgnoreCase(m2.methodName)
+                        && hasSameSignature(m.parameters, m2.parameters)) {
+                    m.methodComment = m2.methodComment;
+                    found = true;
+                }
+            }
+            if (!found) {
+                System.out.println("not found");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS106)